src/clarity-angular/datagrid/datagrid-hidable-column.directive.ts

```diff
@@ -0,0 +1,121 @@
+import {
+    Directive,
+    TemplateRef,
+    ViewContainerRef,
+    Input,
+    AfterViewInit
+} from "@angular/core";
+
+import { DatagridHideableColumn } from "./datagrid-hideable-column";
+import { DatagridColumn } from "./datagrid-column";
+
+@Directive({
+    selector: "[clrDgHideableColumn]"
+})
+
+/**
+ * @class DatagridHideableColumnDirective
+ *
+ * @description
+ * A structural directive meant to be used inside a clr-dg-column component.
+ *
+ * <clr-dg-column>
+ *       <ng-container *clrDgHideableColumn="{ hidden: true }">
+ *           User ID
+ *       </ng-container>
+ *   </clr-dg-column>
+ *
+ * It sets up state and properties so that columns can be manges for hide/show by a service and an internal
+ * datagrid toggle component.
+ *
+ */
+export class DatagridHideableColumnDirective implements AfterViewInit {
+
+    /**
+     * @property _hidden
+     *
+     * @description
+     * Used to initialize the column with either hidden or visible state.
+     *
+     * @type boolean
+     */
+    private _hidden: boolean;
+
+    /**
+     * @function clrDgHideableColumn
+     *
+     * @description
+     * Setter fn for the @Input with the same name as this structural directive.
+     * It allows the user to pre-configure the column's hide/show state. { hidden: true }
+     * It's more verbose but has more Clarity.
+     *
+     * @default false
+     *
+     * @type object
+     *
+     * @example
+     * *clrDgHideableColumn
+     * *clrDgHideableColumn={hidden: false}
+     * *clrDgHideableColumn={hidden: true}
+     *
+     * @param value
+     *
+     */
+    @Input("clrDgHideableColumn")
+    set clrDgHideableColumn( value: any ) {
+        this._hidden = (value) ? value.hidden : false;
```

Comment:
I think we'll go with this for now. Structural directives inputs are not great right now, so we need to make some compromises.

Maybe if https://github.com/angular/angular/issues/15067 we can look for something simpler. And I can see ways for us to progressively deprecate the current explicit form as to avoid breaking changes.